source/h1_connection.c

```diff
@@ -439,11 +441,18 @@ int aws_h1_stream_activate(struct aws_http_stream *stream) {
     struct aws_http_connection *base_connection = stream->owning_connection;
     struct h1_connection *connection = AWS_CONTAINER_OF(base_connection, struct h1_connection, base);
 
+    bool connection_open;
     bool should_schedule_task = false;
 
     { /* BEGIN CRITICAL SECTION */
         s_h1_connection_lock_synced_data(connection);
 
+        connection_open = connection->synced_data.is_open;
+        if (!connection_open) {
+            s_h1_connection_unlock_synced_data(connection);
```

Comment:
Yep, you are right. I was thinking that the cross thread task will check for the `new_stream_error_code` anyway, but h1 will not. And, it's a little bit weird that we checked the `new_stream_error_code` when we make the request and check it again when we activate the stream. But as I found the PR for activate stream and saw the discussion between you and Jonathon, I guess it's fine...

Reply:
I saw your changes, good work.

man, in retrospect I regret adding the activate() function
wish we'd changed it instead so the stream pointer was delivered via some on_setup callback, instead of being returned in a "partially initialized" state from make_request() :|
it would have been a bigger change at the time, but we wouldn't have all these weird edge cases